src/Symfony/Component/Validator/Util/PropertyPath.php

```diff
@@ -42,7 +42,7 @@ public static function append($basePath, $subPath)
                 return $basePath.$subPath;
             }
 
-            return $basePath ? $basePath.'.'.$subPath : $subPath;
+            return '' !== (string) $basePath ? $basePath.'.'.$subPath : $subPath;
```

Comment:
if base path is not a string, then the method would also not return a string. so it must be a string.